Feature/dietitians page revemp

kutagorn opened the pull request:
As discussed I changed the dietitians page. There are some new stuff that I added and some things I opted out.

The thing I added was (apart from the UI/UX vision because I did the whole page me self) If the dietitian has more then 2 qualification now there is a +N counter box that shows how much more qualifications they have and if you hover on to it It will show all of their qualifiactions
![image](https://github.com/user-attachments/assets/db28e947-5bfb-475e-aa2f-23c61f75a234)

And the thing I opted out is that I left the right and left side empty by choice we can add ads there and frankly I like the empty space there makes the page feel more aired/fresh idk. 

I need 3 fields from db to dietitians;
available: boolean;
next_available_date?: string;
online_booking?: boolean;

currently I implemented these changes so when those fields are added to the db they should work live (Hopefully)

Mobile site right now is not ideal at all, I need to take a look but that is going to take too much time. 

Filters are working you can select more then one and it will show you the dietitian that have those 2 qualifications. 
![image](https://github.com/user-attachments/assets/5b1e5ee7-d7ff-4348-b62f-eb82ac6c7309)
I mean I know that is what to be expected from a filter but It was cancer to add so I wanted to share.

Some pixels here and there might not look ideal but they are going to be miniscule changes so, I want this PR to be merged so I can do the other stuff and not get conflicts when I try to merge everything. 

Tests and build works, on my end at least in my local the 404 issue was not here, I couldnt replicate it so we will see if this fixes it but I doubt it cause the page is still static. Please merge.

ozgurozbek commented:
> If the dietitian has more then 2 qualification now there is a +N counter box

This is nice.

> right and left side empty by choice we can add ads there

This is for ad providers to decide, which will infiltrate the content area usually. To do this, that means you'll need to talk to an ad provider service with a custom thing you've done.

> Mobile site right now is not ideal at all
> but that is going to take too much time

Why should we merge before it's responsive? Why is it going to take too much time? Are you not working with responsiveness in mind?

> Filters are working

That's nice! Does +N thing impact this search?

> Some pixels here and there might not look ideal

Mel says your paddings are all wrong as well to be fair. Please fix before merge.

> I can do the other stuff and not get conflicts

Merge this PR to whatever changes you plan on doing before working on them. I'm not okay with taking this live.

> I couldnt replicate it

I can see you haven't built the app. You need to build the app and run the built version to test the 404 issue. This is also not related to this PR, please open a new PR for this issue.

ozgurozbek reviewed (changes requested):
https://github.com/nutrition-ist/fit-nutrition-fe/pull/16#issuecomment-2845883752

ozgurozbek commented:
@hazalmaslan This is for you to decide on, I really don't know what the wenv looks like and you two talking is easier. See requested changes comment.

kutagorn commented:
I split the old FilterBar component in two, Dekstop and Mobile. Desktop is exactly the same (with the fixes)

for Mobile I used SwipeableDrawer, I think it looks good now. 
![image](https://github.com/user-attachments/assets/7b3a5061-abcc-4f16-82c0-77dc8f8e7864)
![image](https://github.com/user-attachments/assets/24b6162a-e997-4ab0-9740-846fd6d51741)

kutagorn commented:
> That's nice! Does +N thing impact this search?
Nope! We already have everything, I am just not showing it. So filter works. That is how I figured the issue and add the +N btw, filter was calling people without that qualification then, I dug deeper and found out those people had the qualifiaction but It didn't showed because I restrected the qualification area for 2 lines. 

> Mel says your paddings are all wrong as well to be fair. Please fix before merge.
Paddings and the white spaces on the sides are fixed, I increased the dietitian count on the page, trust me this looks much better then 4 big ones. 
![image](https://github.com/user-attachments/assets/4552d9c9-b645-4339-9fbb-2276f4ccd4b6)


> Also looks like you've removed the last position on going back to this page which was a feature we were all happy with, or I misread the section in the code. Could you clarify?
I was going to improve this by adding our filters to it aswell, so the site was going to remember your filter choices. This was one of the things I was going to do later. I added the going back to the page back and added the functionality I wanted as well, guess later was today. 

> Merge this PR to whatever changes you plan on doing before working on them. I'm not okay with taking this live.
All of my plans are complete.


Build and Test works, mobile works

ozgurozbek commented:
Sounds great. Future conversations should happen on the next PRs. I also apologise if I came across a bit harsh. Happy with this, but will require iteration mostly by Mel's feedback.
